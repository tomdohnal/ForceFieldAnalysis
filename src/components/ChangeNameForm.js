// @flow
import React from 'react';
import { Form, Input, Button } from 'semantic-ui-react';

import { moveCaretAtTheEnd } from '../helpers';

type Props = {
  onSubmit: () => void,
  onInputChange: (e: SyntheticEvent<HTMLInputElement>, value: { value: string }) => void,
  inputValue: ?string,
  newlyDisplayed: boolean,
}

const ChangeNameForm = ({
  onSubmit, onInputChange, inputValue, newlyDisplayed,
}: Props) => (
  <Form onSubmit={onSubmit}>
    <Form.Group>
      <Input
        size="massive"
        style={{ width: '100%' }}
        onChange={onInputChange}
        placeholder="Name of your change"
        value={inputValue}
        onFocus={moveCaretAtTheEnd}
        autoFocus={newlyDisplayed}
      />
    </Form.Group>
    <Form.Group>
      <Button>Submit Name</Button>
    </Form.Group>
  </Form>
);

export default ChangeNameForm;
