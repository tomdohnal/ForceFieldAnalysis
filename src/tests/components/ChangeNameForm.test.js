import React from 'react';
import { shallow } from 'enzyme';

import ChangeNameForm from '../../components/ChangeNameForm';

describe('ChangeNameForm', () => {
  const onSubmit = jest.fn();
  const onInputChange = jest.fn();

  const ShallowChangeNameForm = shallow(
    <ChangeNameForm
      inputValue="the name of the force"
      onSubmit={onSubmit}
      onInputChange={onInputChange}
    />,
  );

  describe('rendering', () => {
    it('renders correctly', () => {
      expect(ShallowChangeNameForm).toMatchSnapshot();
    });
  });

  describe('functions triggering', () => {
    it('fires onSubmit function when the form is submitted', () => {
      ShallowChangeNameForm.dive().find('form').simulate('submit');

      expect(onSubmit).toBeCalled();
    });

    it('fires onInputChange event when the input is changed', () => {
      ShallowChangeNameForm.find('Input').dive().find('input').simulate('change');

      expect(onInputChange).toBeCalled();
    });
  });
});
