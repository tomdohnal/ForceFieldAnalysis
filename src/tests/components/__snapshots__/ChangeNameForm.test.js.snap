// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChangeNameForm rendering renders correctly 1`] = `
<Form
  as="form"
  onSubmit={[Function]}
>
  <FormGroup>
    <Input
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="Name of your change"
      size="massive"
      style={
        Object {
          "width": "100%",
        }
      }
      type="text"
      value="the name of the force"
    />
  </FormGroup>
  <FormGroup>
    <Button
      as="button"
    >
      Submit Name
    </Button>
  </FormGroup>
</Form>
`;
